services/static-webserver/client/source/class/osparc/file/FileLabelWithActions.js

```diff
@@ -116,40 +160,58 @@ qx.Class.define("osparc.file.FileLabelWithActions", {
       return null;
     },
 
-    // Request to the server and download
-    __retrieveURLAndDownload: function() {
-      let selection = this.getItemSelected();
-      if (selection) {
-        const fileId = selection.getFileId();
-        const locationId = selection.getLocation();
-        osparc.utils.Utils.retrieveURLAndDownload(locationId, fileId)
-          .then(data => {
-            if (data) {
-              osparc.DownloadLinkTracker.getInstance().downloadLinkUnattended(data.link, data.fileName);
-            }
-          });
+    __retrieveURLAndDownloadSelected: function() {
+      if (this.isMultiSelect()) {
+        this.__multiSelection.forEach(selection => {
+          this.__retrieveURLAndDownloadFile(selection);
+        });
+      } else {
+        const selection = this.getItemSelected();
+        if (selection) {
+          this.__retrieveURLAndDownloadFile(selection);
+        }
       }
     },
 
-    __deleteFile: function() {
-      let selection = this.getItemSelected();
-      if (selection) {
-        console.log("Delete ", selection);
-        const fileId = selection.getFileId();
-        const locationId = selection.getLocation();
-        if (locationId !== 0 && locationId !== "0") {
-          osparc.FlashMessenger.getInstance().logAs(this.tr("Only files in simcore.s3 can be deleted"));
-          return false;
+    __deleteSelected: function() {
+      if (this.isMultiSelect()) {
+        this.__multiSelection.forEach(selection => {
+          this.__deleteFile(selection);
```

Comment:
I guess that this does not wait for the files to be deleted, right?

If the user selects 100/1000 files this means that you will send 100/1000 delete actions. I don't think this is a good idea. Could we maybe batch? Let's say 10 max at a time?

Reply:
There is a pending task to support batch operations
https://github.com/ITISFoundation/osparc-issues/issues/1790